src/hooks/useLocalizedConstants.ts

```diff
@@ -27,9 +30,24 @@ export const useLocalizedConstants = () => {
   );
 
   const getLocalizedPhase = useCallback(
-    (phaseEnum: PhaseEnum | null | undefined): string => {
+    (phaseEnum: PhaseEnum | string | null | undefined): string => {
+      if (
+        phaseEnum === null ||
+        phaseEnum === undefined ||
+        !Object.values<string>(PhaseEnum).includes(phaseEnum)
+      ) {
+        return '';
+      }
+      return phasesMap.get(phaseEnum as PhaseEnum)?.translatedName || '';
+    },
+    [constants],
+  );
```

Comment:
You could rewrite the whole function as this. I haven't tested this to confirm. But you do need to switch out the dependency `constants` for `phasesMap`.
```suggestion
    const getLocalizedPhase = useCallback(
    (phaseEnum: PhaseEnum | string | null | undefined): string => {
      return phasesMap.get(phaseEnum as PhaseEnum)?.translatedName || '';
    },
    [phasesMap],
  );
```